src/InMemoryRepository.php

```diff
@@ -256,11 +269,55 @@ private function removeResource(PuliResource $resource)
 
         unset($this->resources[$path]);
 
+        if ($resource instanceof LinkResource && array_key_exists($path, $this->links)) {
```

Comment:
`array_key_exists()` is a function, `isset()` is a language construct. They behave slightly differently (`isset()` returns `false` for `null` values while `array_key_exists()` returns `true`) but that doesn't matter here.